internal/payloads/error_payload.go

```diff
@@ -67,40 +69,117 @@ func PubkeyDuplicateError(ctx context.Context, message string, err error) *Respo
 	}
 }
 
-func ClientError(ctx context.Context, client string, message string, err error, status int) *ResponseError {
-	msg := fmt.Sprintf("%v error: %s: %v", client, message, err)
+func ClientError(ctx context.Context, err error) *ResponseError {
+	var status int
+	var message string
+	switch err {
+	case clients.NotFoundErr:
+		status = 404
+		message = "service returned not found or no data"
+	case clients.UnauthorizedErr:
+		status = 401
+		message = "service returned unauthorized"
+	case clients.ForbiddenErr:
+		status = 403
+		message = "service returned forbidden"
+	case clients.Non2xxResponseErr:
+		status = 500
+		message = "service did not return 2xx"
+	case clients.UnknownAuthenticationTypeErr:
+		status = 500
+		message = "unknown authentication type"
+	case clients.ClientErr:
+		status = 500
+		message = "unknown client error"
+	default:
+		status = 500
+		message = "unknown client error"
+	}
+
 	if logger := ctxval.Logger(ctx); logger != nil {
-		logger.Error().Msg(msg)
+		logger.Error().Msg(message)
 	}
 	return &ResponseError{
 		HTTPStatusCode: status,
-		Message:        msg,
+		Message:        message,
+		TraceId:        ctxval.TraceId(ctx),
+		Error:          err,
+	}
+}
+
+func SourcesError(ctx context.Context, err error) *ResponseError {
+	var status int
+	var message string
+	switch err {
+	case httpClients.MoreThanOneAuthenticationForSourceErr:
+		status = 500
+		message = "more than one authentication"
+	case httpClients.ApplicationNotFoundErr:
+		status = 404
+		message = "application not found"
+	case httpClients.ApplicationTypeNotFoundErr:
+		status = 404
+		message = "unexpected source type"
+	case httpClients.SourceNotFoundErr:
+		status = 404
+		message = "source not found"
+	case httpClients.AuthenticationSourceAssociationErr:
+		status = 500
+		message = "authentication associated to source id not found"
+	case httpClients.AuthenticationForSourcesNotFoundErr:
+		status = 404
+		message = "authentication for source not found"
+	case httpClients.SourcesErr:
+		status = 500
+		message = "unknown sources error"
+	default:
+		status = 500
+		message = "unknown sources error"
+	}
+	if logger := ctxval.Logger(ctx); logger != nil {
+		logger.Error().Msg(message)
+	}
+	return &ResponseError{
+		HTTPStatusCode: status,
+		Message:        message,
+		TraceId:        ctxval.TraceId(ctx),
+		Error:          err,
+	}
+}
```

Comment:
I think you can take advantage of wrapping and have just a single payload function for all client errors:

* common client errors
* sources client errors
* image builder client errors (to be done)

One switch statement could find which one is it out of the three, the other one can then decide the message and code.